### PR TITLE
[Bugfix][CI] ALiBi test case in xformers multi_query_kv_attention

### PR DESCRIPTION
**Pull request by NickLucche**
I've been meaning to add a test case to address this TODO https://github.com/vllm-project/vllm/blob/main/tests/kernels/test_attention.py#L367 but I then realized the test would break due to a dim expansion of the attention bias happening here https://github.com/vllm-project/vllm/blob/main/vllm/attention/backends/xformers.py#L783.

Therefore, this PR adds a test case for xformers multi query+alibi bias and fixes the previously untested scenario.

### CODE REVIEW

File: tests/kernels/test_attention.py [hunk 1]
```diff
@@ -364,10 +372,10 @@ def ref_multi_query_kv_attention(
     return torch.cat(ref_outputs, dim=0)
 
 
-# TODO(woosuk): Add tests for USE_ALIBI=True.
 @pytest.mark.parametrize("num_seqs", NUM_PREFILL_SEQS)
 @pytest.mark.parametrize("num_heads", NUM_HEADS)
 @pytest.mark.parametrize("head_size", HEAD_SIZES)
+@pytest.mark.parametrize("use_alibi", USE_ALIBI)
```
Comment (simon-mo) [inline tests/kernels/test_attention.py:378]: This does increase the test time by 2x because it's testing all the possible combinations. Can we make a separate test case for alibi and test a (very) smalls subset of shapes?

**Review by simon-mo — COMMENTED** [review]
Thank you for the changes!

File: tests/kernels/test_attention.py (hunk 1 above)
Comment (NickLucche) [reply tests/kernels/test_attention.py:378]: sure thing

**Comment by NickLucche** [comment]
Thanks for reviewing!

**Review by robertgshaw2-redhat — APPROVED** [review]

**Comment by DarkLight1337** [comment]
Can you merge from main to see if the test failures are still relevant?

**Comment by NickLucche** [comment]
This was conflicting with the recent `typing.List` change. Could you give it another try at merging? Thanks!

**Comment by DarkLight1337** [comment]
Can you fix the pre-commit failure?

**Comment by NickLucche** [comment]
yeah sorry a `Tuple` managed to slip through
